OneTimePasswordTests/TokenPersistenceTests.swift

```diff
@@ -26,229 +26,190 @@
-                XCTFail("Failed to recover persistent token with identifier: \(keychainItem.identifier)")
-                return
+            // Fetch the token again
+            do {
+                let fetchedToken = try keychain.persistentTokenWithIdentifier(savedToken.identifier)
+                XCTAssertEqual(fetchedToken?.token, modifiedToken)
+                XCTAssertEqual(fetchedToken?.identifier, savedToken.identifier)
+            } catch {
+                XCTFail("persistentTokenWithIdentifier(_:) failed with error: \(error)")
             }
-            XCTAssertEqual(thirdKeychainItem.token, modifiedToken)
-            XCTAssertEqual(thirdKeychainItem.identifier, keychainItem.identifier)
-        } catch {
-            XCTFail("persistentTokenWithIdentifier(_:) failed with error: \(error)")
-            return
-        }
 
-        // Remove the token
-        do {
-            try keychain.deletePersistentToken(keychainItem)
-        } catch {
-            XCTFail("deletePersistentToken(_:) failed with error: \(error)")
-            return
-        }
+            // Remove the token
+            do {
+                try keychain.deletePersistentToken(savedToken)
+            } catch {
+                XCTFail("deletePersistentToken(_:) failed with error: \(error)")
+            }
 
-        // Attempt to restore the deleted token
-        do {
-            let fourthKeychainItem = try keychain.persistentTokenWithIdentifier(keychainItem.identifier)
-            XCTAssertNil(fourthKeychainItem)
+            // Attempt to restore the deleted token
+            do {
+                let fetchedToken = try keychain.persistentTokenWithIdentifier(savedToken.identifier)
+                XCTAssertNil(fetchedToken, "Token should have been removed from keychain")
+            } catch {
+                XCTFail("persistentTokenWithIdentifier(_:) failed with error: \(error)")
+            }
         } catch {
-            XCTFail("persistentTokenWithIdentifier(_:) failed with error: \(error)")
-            return
+            XCTFail("addToken(_:) failed with error: \(error)")
         }
     }
 
-    func testDuplicateURLs() {
-        guard let token1 = Token.URLSerializer.deserialize(kValidTokenURL),
-            let token2 = Token.URLSerializer.deserialize(kValidTokenURL) else {
-                XCTFail("Failed to construct tokens from url.")
-                return
-        }
-
-        // Add both tokens to the keychain
-        guard let savedItem1 = try? keychain.addToken(token1) else {
-            XCTFail("Failed to save to keychain: \(token1)")
-            return
-        }
-        guard let savedItem2 = try? keychain.addToken(token2) else {
-            XCTFail("Failed to save to keychain: \(token2)")
-            return
-        }
-        XCTAssertEqual(savedItem1.token, token1)
-        XCTAssertEqual(savedItem2.token, token2)
+    func testDuplicateTokens() {
```

Comment:
Length Violation: Function body should be span 40 lines or less: currently spans 70 lines